pkg/container-collection/options.go

```diff
@@ -82,6 +83,27 @@ func containerRuntimeEnricher(
 		return true
 	}
 
+	// Enrich labels when we have a sandbox ID. cri-o might not find the corresponding container yet
+	if container.SandboxId != "" {
+		criClient, ok := runtimeClient.(*cri.CRIClient)
+		if ok {
+			labels, err := criClient.GetPodLabels(container.SandboxId)
+			if err != nil {
+				log.Printf("Runtime enricher (%s): failed to GetPodLabels: %s",
+					runtimeName, err)
+				// Container could be managed by another runtime, don't drop it.
+				return true
+			}
+			log.Printf("Runtime enricher (%s): pod labels: %v", runtimeName, labels)
+			container.K8s.PodLabels = labels
+
+			// If this was the only information needed, skip rest of this enricher
+			if runtimeclient.IsEnrichedWithK8sMetadata(container.K8s.BasicK8sMetadata) {
+				return true
+			}
+		}
+	}
```

Comment:
Actually, I don't understand why is all this necessary if the pod labels will be available in the `ContainerData` struct returned by the `GetContainer` call below. In fact, it is breaking all the idea of using an abstract `runtimeClient` that doesn't know/care about the actual runtime or protocol used.

Reply:
See comment https://github.com/inspektor-gadget/inspektor-gadget/pull/2398#issuecomment-1945366447.